datetime.go

```diff
@@ -283,3 +283,214 @@ func (dt *DateTime) NotInRange(min, max time.Time) *DateTime {
 
 	return dt
 }
+
+// Zone returns a new String instance with datetime zone.
+//
+// Example:
+//
+//	dt := NewDateTime(t, time.Unix(0, 10))
+//	dt.Zone().Equal("/foo")
+func (dt *DateTime) Zone() *String {
+	dt.chain.enter("Zone()")
+	defer dt.chain.leave()
+
+	if dt.chain.failed() {
+		return newString(dt.chain, "")
+	}
+
+	zone, _ := dt.value.Zone()
+	return newString(dt.chain, zone)
+}
+
+// Year returns the year in which datetime occurs.
+//
+// Example:
+//
+//	dt := NewDateTime(t, time.Unix(0, 0))
+//	dt.Year().Equal(float64(1970))
+func (dt *DateTime) Year() *Number {
+	dt.chain.enter("Year()")
+	defer dt.chain.leave()
+
+	if dt.chain.failed() {
+		return newNumber(dt.chain, float64(0))
+	}
+
+	return newNumber(dt.chain, float64(dt.value.Year()))
+}
+
+// Month returns the month of the year specified by datetime.
+//
+// Example:
+//
+//	dt := NewDateTime(t, time.Unix(0, 0))
+//	dt.Month().Equal(float64(1))
+func (dt *DateTime) Month() *Number {
+	dt.chain.enter("Month()")
+	defer dt.chain.leave()
+
+	if dt.chain.failed() {
+		return newNumber(dt.chain, float64(0))
+	}
+
+	return newNumber(dt.chain, float64(dt.value.Month()))
+}
+
+// Day returns the day of the month specified datetime.
+//
+// Example:
+//
+//	dt := NewDateTime(t, time.Unix(0, 0))
+//	dt.Day().Equal(float64(1))
+func (dt *DateTime) Day() *Number {
+	dt.chain.enter("Day()")
+	defer dt.chain.leave()
+
+	if dt.chain.failed() {
+		return newNumber(dt.chain, float64(0))
+	}
+
+	return newNumber(dt.chain, float64(dt.value.Day()))
+}
+
+// Weekday returns the day of the week specified by datetime.
```

Comment:
Let's also specify the range of returned value.

Reply:
Done